test/e2e/auth/service_accounts.go

```diff
@@ -621,6 +624,92 @@ var _ = SIGDescribe("ServiceAccounts", func() {
 		framework.ExpectNoError(podErr)
 		framework.Logf("completed pod")
 	})
+
+	ginkgo.It("should run through the lifecycle of a ServiceAccount", func() {
+		testNamespaceName := f.Namespace.Name
+		testServiceAccountName := "testserviceaccount"
+		testSecretName := "testsecret"
+		testServiceAccountStaticLabels := map[string]string{"test-serviceaccount-static": "true"}
+		testServiceAccountStaticLabelsFlat := "test-serviceaccount-static=true"
+
+		ginkgo.By("creating a ServiceAccount")
+		testServiceAccount := v1.ServiceAccount{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   testServiceAccountName,
+				Labels: testServiceAccountStaticLabels,
+			},
+		}
+		_, err := f.ClientSet.CoreV1().ServiceAccounts(testNamespaceName).Create(context.TODO(), &testServiceAccount, metav1.CreateOptions{})
+		framework.ExpectNoError(err, "failed to create a ServiceAccount")
+		ginkgo.By("creating a Secret")
+		testSecret := v1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: testSecretName,
+			},
+			Data: map[string][]byte{
+				"test-field": []byte("test-value"),
+			},
+			Type: "Opaque",
+		}
+		_, err = f.ClientSet.CoreV1().Secrets(testNamespaceName).Create(context.TODO(), &testSecret, metav1.CreateOptions{})
+		framework.ExpectNoError(err, "failed to create a Secret")
+
+		fmt.Println("watching for the ServiceAccount to be added")
+		resourceWatchTimeoutSeconds := int64(180)
+		resourceWatch, err := f.ClientSet.CoreV1().ServiceAccounts(testNamespaceName).Watch(context.TODO(), metav1.ListOptions{LabelSelector: testServiceAccountStaticLabelsFlat, TimeoutSeconds: &resourceWatchTimeoutSeconds})
+		if err != nil {
+			fmt.Println(err, "failed to setup watch on newly created ServiceAccount")
+			return
+		}
+
+		resourceWatchChan := resourceWatch.ResultChan()
+		for watchEvent := range resourceWatchChan {
+			if watchEvent.Type == watch.Added {
+				break
+			}
+		}
+
+		defer func() {
+			ginkgo.By("deleting the ServiceAccount")
+			err = f.ClientSet.CoreV1().ServiceAccounts(testNamespaceName).DeleteCollection(context.TODO(), metav1.DeleteOptions{}, metav1.ListOptions{})
```

Comment:
It would probably useful to directly exercise the `Delete` method here as well, rather than have that method implicitly covered by other tests

Reply:
I'm not sure where this would fit, as `DeleteCollection` would have removed the ServiceAccount.
In order to do this, a new ServiceAccount would need to be created. Would it be worth putting that in?